kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+package edu.arizona.kfs.fp.service.impl;
+
+import static org.kuali.kfs.sys.KFSConstants.GL_CREDIT_CODE;
+import static org.kuali.kfs.sys.KFSConstants.GL_DEBIT_CODE;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.kuali.kfs.coa.businessobject.ObjectCode;
+import org.kuali.kfs.coa.service.ObjectCodeService;
+import org.kuali.kfs.integration.purap.PurchasingAccountsPayableModuleService;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.businessobject.Bank;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntry;
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySequenceHelper;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.document.AccountingDocument;
+import org.kuali.kfs.sys.document.GeneralLedgerPostingDocument;
+import org.kuali.kfs.sys.document.service.AccountingDocumentRuleHelperService;
+import org.kuali.kfs.sys.document.validation.impl.AccountingDocumentRuleBaseConstants.GENERAL_LEDGER_PENDING_ENTRY_CODE;
+import org.kuali.kfs.sys.service.BankService;
+import org.kuali.kfs.sys.service.GeneralLedgerPendingEntryService;
+import org.kuali.kfs.sys.service.NonTransactional;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import edu.arizona.kfs.fp.businessobject.PaymentMethod;
+import edu.arizona.kfs.fp.businessobject.PaymentMethodChart;
+import edu.arizona.kfs.fp.service.PaymentMethodGeneralLedgerPendingEntryService;
+
+
+@NonTransactional
+public class PaymentMethodGeneralLedgerPendingEntryServiceImpl implements PaymentMethodGeneralLedgerPendingEntryService {
+    private static final Logger LOG = Logger.getLogger(PaymentMethodGeneralLedgerPendingEntryServiceImpl.class);
+
+    protected static final String DEFAULT_PAYMENT_METHOD_IF_MISSING = "A"; // check/ACH
+    protected static final String PRNC = "PRNC";
+    protected static final String CMNC = "CMNC";
+    protected static final String AP_CREDIT_CARD_BANK_CODE = "0007";
+    protected static final String AP_CREDIT_CARD_PAYMENT_METHOD_CODE = "C";
+    
+    protected GeneralLedgerPendingEntryService generalLedgerPendingEntryService;
+    protected ObjectCodeService objectCodeService;
+    protected ParameterService parameterService;
```

Comment:
I realized that parameter service is not actually being used in this class, so let's remove the import, property and getter/setter methods.